src/value/number.rs

```diff
@@ -0,0 +1,483 @@
+    /// # extern crate serde_yaml;
+    /// #
+    /// # use std::i64;
+    /// #
+    /// # fn yaml(i: &str) -> serde_yaml::Value { serde_yaml::from_str(i).unwrap() }
+    /// # fn main() {
+    /// let big = i64::MAX as u64 + 10;
+    /// let v = yaml(r#"
+    /// ---
+    /// a: 64
+    /// b: 9223372036854775817
+    /// c: 256.0
+    /// "#);
+    ///
+    /// assert_eq!(v["a"].as_i64(), Some(64));
+    /// assert_eq!(v["b"].as_i64(), None);
+    /// assert_eq!(v["c"].as_i64(), None);
+    /// # }
+    /// ```
+    #[inline]
+    pub fn as_i64(&self) -> Option<i64> {
+        match self.n {
+            N::PosInt(n) => NumCast::from(n),
+            N::NegInt(n) => Some(n),
+            N::Float(_) => None,
+        }
+    }
+
+    /// If the `Number` is an integer, represent it as u64 if possible. Returns
+    /// None otherwise.
+    ///
+    /// ```rust
+    /// # #[macro_use]
+    /// # extern crate serde_yaml;
+    /// #
+    /// # fn yaml(i: &str) -> serde_yaml::Value { serde_yaml::from_str(i).unwrap() }
+    /// # fn main() {
+    /// let v = yaml(r#"
+    /// ---
+    /// a: 64
+    /// b: -64
+    /// c: 256.0
+    /// "#);
+    ///
+    /// assert_eq!(v["a"].as_u64(), Some(64));
+    /// assert_eq!(v["b"].as_u64(), None);
+    /// assert_eq!(v["c"].as_u64(), None);
+    /// # }
+    /// ```
+    #[inline]
+    pub fn as_u64(&self) -> Option<u64> {
+        match self.n {
+            N::PosInt(n) => Some(n),
+            N::NegInt(n) => NumCast::from(n),
+            N::Float(_) => None,
+        }
+    }
+
+    /// Represents the number as f64 if possible. Returns None otherwise.
+    ///
+    /// ```rust
+    /// # #[macro_use]
+    /// # extern crate serde_yaml;
+    /// #
+    /// # fn main() {
+    /// # fn yaml(i: &str) -> serde_yaml::Value { serde_yaml::from_str(i).unwrap() }
+    /// let v = yaml(r#"
+    /// ---
+    /// a: 256.0
+    /// b: 64
+    /// c: -64
+    /// "#);
+    ///
+    /// assert_eq!(v["a"].as_f64(), Some(256.0));
+    /// assert_eq!(v["b"].as_f64(), Some(64.0));
+    /// assert_eq!(v["c"].as_f64(), Some(-64.0));
+    /// # }
+    /// ```
+    /// ```rust
+    /// # use std::f64;
+    /// # fn yaml(i: &str) -> serde_yaml::Value { serde_yaml::from_str(i).unwrap() }
+    /// assert_eq!(yaml("inf").as_f64(), Some(f64::INFINITY));
+    /// assert_eq!(yaml("-inf").as_f64(), Some(f64::NEG_INFINITY));
+    /// assert!(yaml("NaN").as_f64().unwrap().is_nan());
+    /// ```
+    #[inline]
+    pub fn as_f64(&self) -> Option<f64> {
+        match self.n {
+            N::PosInt(n) => NumCast::from(n),
+            N::NegInt(n) => NumCast::from(n),
+            N::Float(n) => Some(n),
+        }
+    }
+
+    /// Converts a finite `f64` to a `Number`. Infinite or NaN values are YAML
+    /// numbers.
+    ///
+    /// ```rust
+    /// # use std::f64;
+    /// #
+    /// # use serde_yaml::Number;
+    /// #
+    /// assert!(Number::from_f64(256.0).is_some());
+    ///
+    /// //NAN is a number in YAML
```

Comment:
Sorry, I don't think these comments add anything, the code basically says just that.